docs/development/build-instructions-gn.md

```diff
@@ -38,12 +47,56 @@ $ gclient sync --with_branch_heads --with_tags
 ```sh
 $ cd src
 $ export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
-$ gn gen out/Default --args='root_extra_deps=["//electron"] is_electron_build=true is_component_build=true use_jumbo_build=true v8_promise_internal_field_count=1 v8_typed_array_max_size_in_heap=0'
+$ gn gen out/Default --args='target_os="mac"'
 ```
 
-This will generate all the ninja files needed for the build. You shouldn't have
-to run `gn gen` again—if you want to change the build arguments you can run `gn
-args out/Default` to bring up an editor.
+This will generate a build directory `out/Default` under `src/` with
+default chromium build configuration. You can replace `Default` with
+another name, but it should be a subdirectory of `out`. Here `target_os`
+indicates the primary operating system used for the build, you can run
+`gn help target_os` to get list of possible values. Also, to know the list
+of available configuration options, run `gn args out/Default --list`.
+To build `electron` we need some additional arguments depending on the build,
+which are specified below. Also you shouldn't have to run `gn gen` again—if
+you want to change the build arguments, you can run `gn args out/Default` to
+bring up an editor.
+
+* #### GN args for Debug/Component build of Electron:
+
+```gn
+root_extra_deps=["//electron"]
+is_electron_build=true
+is_component_build=true
+use_jumbo_build=true
+is_debug=true
+symbol_level=2
+use_allocator="none"
```

Comment:
why not leave this default?

Reply:
On linux other than arm arch, it defaults to `tcmalloc` which has some issues with running inside docker. The current gyp system doesn't work with it https://github.com/electron/libchromiumcontent/blob/master/chromiumcontent/args/shared_library.gn#L7. I haven't verified with gn yet.

Reply:
it seems to work fine on circleci without this, which i believe is run inside docker.